src/twisted/web/test/requesthelper.py

```diff
@@ -353,7 +352,7 @@ def setHost(self, host, port, ssl=0):
         if port == default:
```

Comment:
This comment applies to all the `intToBytes` conversions -- is this safe?  Specifically, intToBytes does provide some validation of the input where these changes will not.  For example, with trunk on both python2 and 3, the following:

```
from twisted.web.http import HTTPChannel, Request

ch = HTTPChannel()
r = Request(ch)
r.setHost(b'http://www.example.com', 'abc')
```
Will raise a TypeError, however with these changes, no error will be raised.

Reply:
There is not a lot of input validation done on all these arguments to begin with.
I added https://github.com/twisted/twisted/pull/911/commits/38ee5f3ca983c486521c9cffad5cb8afdbc3f54a which will fail on the testcase you provided.

Reply:
Thanks!  It was primarily the change in validation that drew my attention, not that I was expecting your changes to add new validation.